### DISCUSSION TITLE
I want to connect Gnosis Safe to Uniswap using wallet connect, help me understand the process

### DISCUSSION THREAD

**Discussion by bhavya2611** [WalletConnect/walletconnect-monorepo#2736] [Developer Support] [answered]
I want to connect Gnosis Safe to Uniswap using wallet connect, please guide me on what are the steps or to the right piece of code / example for this

**Comment by EthanBlockson** [comment, marked as answer]
Your Safe is a contract, not just an address.

To interact with Uniswap contracts from Safe, follow these steps:

1) Press New transaction > Contract interaction (in the Safe)
2) Paste the Uniswap contract address and ABI
3) Choose the contract call you want to perform

**Reply by 0xDamned** [reply]
Good!

**Reply by AndreyPetrov74** [reply]
thank you for useful answer

**Reply by Variant101** [reply]
oh, thank you bro

**Reply by Luchia777** [reply]
Helpful. Thank you.

**Reply by Svetik0102** [reply]
Thank  you

**Comment by spicedoubles** [comment]
It is interested

**Comment by jendos71** [comment]
Needed to get gitcoin passport.

**Reply by berkovboris** [reply]
oh yeah

**Reply by berkovboris** [reply]
follow me

**Comment by jendos71** [comment]
Good project.

**Comment by prsm171** [comment]
Good

**Comment by Tittanc** [comment]
nice nice

**Comment by 1VoV4ik1** [comment]
I love this exchanger))

**Reply by LunaLi777** [reply]
Me too.

**Comment by Rusgreat** [comment]
Fantastic!

**Comment by FlameRow** [comment]
Good

**Comment by Kasian8702** [comment]
thanks, was the same task

**Comment by Achyt04** [comment]
very ,very good)

**Comment by ksctyrj56** [comment]
thanks!

**Comment by 1VoV4ik1** [comment]
Hi gues))

**Comment by Brandon0x** [comment]
Great.

**Comment by MaryTop1** [comment]
Thanks for the info.

**Comment by naturalspirit299** [comment]
Thank you for reaching out and showing interest in connecting Gnosis Safe to Uniswap. It's great to see you exploring the integration possibilities. The process can be a bit complex, but don't worry, I'm here to help! Let's break it down step by step and ensure you have a seamless experience. Feel free to ask any questions along the way!

**Comment by novicjen** [comment]
Useful information for a beginner

**Comment by Denis91Rodman** [comment]
I also want to know how to do it

**Comment by RonHarper9** [comment]
I want it too, man

**Comment by Charlotta23** [comment]
u neeed to buy gnosis first

**Comment by valeriarejn6** [comment]
great project

**Comment by Zahangir1234** [comment]
interesting

**Comment by RonHarper9** [comment]
I like it

**Comment by okavoka** [comment]
Helped me too, thanks!

**Comment by arturbebov45** [comment]
It's cool for me !

**Comment by vikserov14** [comment]
Thank you for this information !

**Comment by Dilimed** [comment]
Good

**Comment by Yuldaza** [comment]
I like. Thanks.
